pkg/controller/nodes/common/utils.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"github.com/lyft/flytepropeller/pkg/apis/flyteworkflow/v1alpha1"
+	"github.com/lyft/flytepropeller/pkg/controller/executors"
+	"github.com/lyft/flytepropeller/pkg/utils"
+	"strconv"
+)
+
+func GenerateUniqueId(parentInfo executors.ImmutableParentInfo, nodeID string) (string, error) {
+	var parentUniqueId v1alpha1.NodeID
+	var parentRetryAttempt string
+
+	if parentInfo != nil {
+		parentUniqueId = parentInfo.GetUniqueID()
+		parentRetryAttempt = strconv.Itoa(int(parentInfo.CurrentAttempt()))
+	}
+
+	return utils.FixedLengthUniqueIDForParts(20, parentUniqueId, parentRetryAttempt, nodeID)
```

Comment:
why 20 chars, it is fine, but, maybe make this a constant?

Reply:
Sure.